plugins/modules/podman_image.py

```diff
@@ -806,19 +799,20 @@ def push_image(self):
             else:
                 dest_format_string = '{transport}:{dest}'
 
-        dest_string = dest_format_string.format(transport=transport, name=self.name, dest=dest, image_name=self.image_name,)
+            dest_string = dest_format_string.format(transport=transport, name=self.name, dest=dest, image_name=self.image_name,)
+        else:
+            if len(dest.split('/')) < 3:
+                dest_string = f'{dest}/{self.image_name}'
+            else:
+                dest_string = dest
```

Comment:
```suggestion
            if dest and len(dest.split('/')) < 3:
                dest_string = f'{dest}/{self.image_name}'
            elif dest:
               dest_string = dest
            else:
                dest_string = name
```

would cover the case where  `dest` isn't set, but `name` is a valid replacement (this is checked on L784-786)